app/connectors/FinancialStatementConnector.scala

```diff
@@ -82,6 +85,23 @@ class FinancialStatementConnector @Inject()(
   }
 
   def getSchemeFS(pstr: String)
+                      (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext, request: RequestHeader): Future[SchemeFS] = {
+    val cacheKey = s"schemeFS-$pstr"
+    cache.getOrElseUpdate[SchemeFS](cacheKey,config.ifsCache){
+      logger.warn(s"Cache missing, reset or not instantiated, fetching getSchemeFS from IFS")
+      getSchemeFSCall(pstr)
+    }.map{ result =>
+      val sizeInBytes = getObjectSize(result)
+      logger.warn(s"Cache hit for getSchemeFS. Item size: $sizeInBytes bytes.")
+      result
+    }
+  }
+
+  private def getObjectSize[T](obj: T)(implicit writes: Writes[T]): Int = {
```

Comment:
I wanted to know what size payloads we would be saving to the cache. Maybe we would need to increase jvm memory.